pkgs/dart_mcp/test/resources_support_test.dart

```diff
@@ -45,9 +46,99 @@ void main() {
           ),
     );
   });
+
+  test('client can subscribe to resource updates from the server', () async {
+    var environment = TestEnvironment(
+      TestMCPClient(),
+      TestMCPServerWithResources.new,
+    );
+    await environment.initializeServer();
+
+    final serverConnection = environment.serverConnection;
+    final server = environment.server;
+
+    final resourceListChangedQueue = StreamQueue(
+      serverConnection.resourceListChanged,
+    );
+
+    final fooResource = Resource(name: 'foo', uri: 'foo://bar');
+    var fooContents = 'bar';
+    server.addResource(
+      fooResource,
+      (_) => ReadResourceResult(
+        contents: [
+          TextResourceContents(uri: fooResource.uri, text: fooContents),
+        ],
+      ),
+    );
+
+    await resourceListChangedQueue.next;
+    final resources = await serverConnection.listResources(
+      ListResourcesRequest(),
+    );
+    expect(
+      resources.resources,
+      unorderedEquals([fooResource, TestMCPServerWithResources.helloWorld]),
+    );
+
+    final resourceChangedQueue = StreamQueue(serverConnection.resourceUpdated);
+    serverConnection.subscribeResource(SubscribeRequest(uri: fooResource.uri));
+    // Let the server process the request
+    await pumpEventQueue();
+
+    fooContents = 'baz';
+    server.updateResource(fooResource);
+
+    expect(
+      await resourceChangedQueue.next,
+      isA<ResourceUpdatedNotification>().having(
+        (n) => n.uri,
+        'uri',
+        fooResource.uri,
+      ),
+    );
+
+    expect(
+      await serverConnection.readResource(
+        ReadResourceRequest(uri: fooResource.uri),
+      ),
+      isA<ReadResourceResult>().having(
+        (r) => r.contents.single,
+        'contents',
+        isA<TextResourceContents>()
+            .having((c) => c.text, 'text', 'baz')
+            .having((c) => c.uri, 'uri', fooResource.uri),
+      ),
+    );
+
+    serverConnection.unsubscribeResource(
+      UnsubscribeRequest(uri: fooResource.uri),
+    );
+    // Let the server process the request
+    await pumpEventQueue();
+
+    fooContents = 'baz';
```

Comment:
I assume we are setting to 'baz' again to show that the resource is not updated if nothing changes? If so, can you add a comment?

Reply:
Oh, this was actually a mistake, although it doesn't actually need to change at all.

There is no change notification due to the unsubscribe a few lines above.